### PR TITLE
records api: batch `get_records`

### PR DESCRIPTION
**Pull request by MJedr**
ref:#2314

### CODE REVIEW

File: backend/inspirehep/records/api/base.py [hunk 1]
```diff
@@ -176,6 +176,17 @@ def get_record(cls, record_uuid, with_deleted=False, record_version=None):
...
         return record
 
+    @classmethod
+    def get_records_batched(cls, ids, with_deleted=False, max_batch=100):
+        for batch in chunker(ids, max_chunk_size=max_batch):
+            query = cls.model_cls.query.filter(cls.model_cls.id.in_(batch))
+            if not with_deleted:
+                query = query.filter(
+                    cls.model_cls.json.op("->")("deleted") == None  # noqa
+                )
```
Comment (michamos) [inline backend/inspirehep/records/api/base.py:186]: this is incorrect, `deleted = False` is allowed in a record. You need to check that `delete: True` doesn't occur.

Comment (MJedr) [reply backend/inspirehep/records/api/base.py:186]: ok, thank you

File: backend/inspirehep/records/api/base.py [hunk 2]
```diff
@@ -176,6 +178,18 @@ def get_record(cls, record_uuid, with_deleted=False, record_version=None):
...
         return record
 
+    @classmethod
+    def get_records_batched(cls, ids, with_deleted=False, max_batch=100):
+        for batch in chunker(ids, max_chunk_size=max_batch):
+            query = cls.model_cls.query.filter(cls.model_cls.id.in_(batch))
+            if not with_deleted:
+                query = query.filter(
+                    (cls.model_cls.json.op("->")("deleted") == None)  # noqa
+                    | (cls.model_cls.json["deleted"] != cast("True", JSONB))
```
Comment (michamos) [inline backend/inspirehep/records/api/base.py:188]: I believe the first line is not needed here, as Postgres puts `NULL` when the key is absent.

**Review by michamos — APPROVED** [review]
Minor comment, looks good :+1:

File: backend/inspirehep/records/api/base.py (hunk 2 above)
Comment (MJedr) [reply backend/inspirehep/records/api/base.py:188]: It is needed, tests don't pass when it's not there (the only result is the record with `deleted:False`)

**Pull request merged**